ozaria/site/components/play/PageIntroLevel/index.vue

```diff
@@ -0,0 +1,176 @@
+      if (!this.campaignId && this.courseId) {
+        const course = await api.courses.get({courseID: this.courseId})
+        const campaignId = course.campaignID
+        await this.$store.dispatch('campaigns/fetch', campaignId)
+        this.campaignData = _.cloneDeep(this.$store.state.campaigns.byId[campaignId])
+      } else if (this.campaignId) {
+        await this.$store.dispatch('campaigns/fetch', this.campaignId)
+        this.campaignData = _.cloneDeep(this.$store.state.campaigns.byId[this.campaignId])
+      }
+
+    } catch (err) {
+      console.error('Error in creating data for intro level', err)
+      noty({text: 'Error in creating data for intro level', type: 'error', timeout: 2000})
+      return
+    }
+    // Assign first content in the sequence to this.currentContent
+    this.currentIndex = 0
+    this.currentContent = this.introContent[this.currentIndex]
+    this.dataLoaded = true
+  },
+  methods: {
+    onContentCompleted: async function () {
+      this.currentIndex++
+      if (this.currentIndex < this.introContent.length) { // increment current content
+        this.currentContent = this.introContent[this.currentIndex]
+      } else { // whole intro content completed
+        await this.setIntroLevelComplete()
+        await this.fetchNextLevel()
+        const link = this.fetchNextLevelLink()
+        if (link) {
+          return application.router.navigate(link, { trigger: true })
+        }
+      }
+    },
+    setIntroLevelComplete: async function () {
+      if (!me.isSessionless()) { // not saving progress/session for teachers
+        try {
+          this.introLevelSession.state.complete = true
+          await api.levelSessions.update(this.introLevelSession)
+        } catch (err) {
+          console.error('Error in saving intro level session', err)
+          return noty({text: 'Error in saving intro level session', type: 'error', timeout: 2000})
+        }
+      }
+    },
+    fetchNextLevel: async function () {
+      try {
+        const currentLevel = this.campaignData.levels[this.introLevelData.original]
+        const nextLevelOriginal = getNextLevelOriginalForLevel(currentLevel)[0]
+        if (nextLevelOriginal)
+          this.nextLevel = await api.levels.getByOriginal(nextLevelOriginal)
+      } catch (err) {
+        console.error("Error in fetching next level", err)
+        noty({text: 'Error in fetching next level', type: 'error', timeout: 2000})
+      }
+    },
+    fetchNextLevelLink: function() {
+      if (this.nextLevel.slug && this.nextLevel.type) {
+        const nextLevelOptions = {
+          courseId: this.courseId,
+          courseInstanceId: this.courseInstanceId,
+          campaignId: this.campaignId
+        }
+        return getNextLevelLink(this.nextLevel, nextLevelOptions)
+      } else {
+        console.log("no next level found") // TODO what to do if last level of campaign
+      }
+    }
+  }
+})
+</script>
+
+<template>
+  <div v-if="dataLoaded">
```

Comment:
Building on the loading discussion above - we already have to handle loading here to communicating progress with events from sub components make make the most sense.